src/DDTrace/ID.php

```diff
@@ -14,6 +19,19 @@ public static function generate()
          * will do. And since all integers in PHP are signed, an int
          * between 1 & PHP_INT_MAX will be 63-bit.
          */
-        return (string) mt_rand(1, PHP_INT_MAX);
+        $first31bits = mt_rand(0, mt_getrandmax()) << 32;
+        $second31bits = mt_rand(0, mt_getrandmax()) << 1;
+        $random1or0 = mt_rand(0, 1);
+        return (string) ($first31bits | $second31bits | $random1or0);
+    }
+
+    /**
+     * The expected max value for the generated id.
+     *
+     * @return int
+     */
+    public static function getMaxId()
+    {
+        return (mt_getrandmax() << 32) | (mt_getrandmax() << 1) | 1;
```

Comment:
I'm curious why this exists. Is it just to show that the return value is the same as `PHP_INT_MAX` in the tests? I don't see it being used anywhere else in the tracer and that test might break on 32-bit boxes. Just double checking. :)